include/boost/process/v2/impl/exe.ipp

```diff
@@ -0,0 +1,324 @@
+{
+    filesystem::path path;
+    if (pid == GetCurrentProcessId()) 
+    {
+        wchar_t buffer[MAX_PATH];
+        if (GetModuleFileNameW(nullptr, buffer, sizeof(buffer)) != 0) 
+        {
+            wchar_t exe[MAX_PATH];
+            if (_wfullpath(exe, buffer, MAX_PATH)) 
+            {
+                path = exe;
+            }
+        }
+    } 
+    else 
+    {
+        HANDLE proc = detail::ext::open_process_with_debug_privilege(pid, ec);
+        if (proc == nullptr) return path;
+        wchar_t buffer[MAX_PATH];
+        DWORD size = sizeof(buffer);
+        if (QueryFullProcessImageNameW(proc, 0, buffer, &size) != 0) 
+        {
+            wchar_t exe[MAX_PATH];
+            if (_wfullpath(exe, buffer, MAX_PATH)) 
+            {
+                path = exe;
+            }
+        }
+        CloseHandle(proc);
+    }
+    if (path.string().empty())
+        ec = detail::get_last_error();
+    return path;
+}
+
+#elif (defined(__APPLE__) && defined(__MACH__))
+
+filesystem::path exe_path(boost::process::v2::pid_type pid, std::error_code & ec)
+{
+    std::string path;
+    char exe[PROC_PIDPATHINFO_MAXSIZE];
+    if (proc_pidpath(pid, exe, sizeof(exe)) > 0) 
+    {
+        char buffer[PATH_MAX];
+        if (realpath(exe, buffer)) 
+        {
+            path = buffer;
+        }
+    }
+    if (path.empty())
+        ec = detail::get_last_error();
+    return path;
+}
+
+#elif (defined(__linux__) || defined(__ANDROID__))
+
+filesystem::path exe_path(boost::process::v2::pid_type pid, std::error_code & ec)
+{
+    std::string path;
+    char exe[PATH_MAX];
+    if (realpath(("/proc/" + std::to_string(pid) + "/exe").c_str(), exe)) 
+    {
+        path = exe;
+    }
+    if (path.empty())
+        ec = detail::get_last_error();
+    return path;
+}
+
+#elif defined(__FreeBSD__) || defined(__DragonFly__)
+
+filesystem::path exe_path(boost::process::v2::pid_type pid, std::error_code & ec) 
+{
+    std::string path;
+    int mib[4]; 
+    std::size_t len;
+    mib[0] = CTL_KERN;
+    mib[1] = KERN_PROC;
+    mib[2] = KERN_PROC_PATHNAME;
+    mib[3] = pid;
+    if (sysctl(mib, 4, nullptr, &len, nullptr, 0) == 0) 
+    {
+        std::string strbuff;
+        strbuff.resize(len, '\0');
+        char *exe = strbuff.data();
+        if (sysctl(mib, 4, exe, &len, nullptr, 0) == 0) 
+        {
+            char buffer[PATH_MAX];
+            if (realpath(exe, buffer)) 
+            {
+                path = buffer;
+            }
+        }
+    }
+    if (path.empty())
+        ec = detail::get_last_error();
+    return path;
+}
+
+#elif defined(__NetBSD__)
+
+filesystem::path exe_path(boost::process::v2::pid_type pid, std::error_code & ec)
+{
+    std::string path;
+    int mib[4]; 
+    std::size_t len;
+    mib[0] = CTL_KERN;
+    mib[1] = KERN_PROC_ARGS;
+    mib[2] = pid;
+    mib[3] = KERN_PROC_PATHNAME;
+    if (sysctl(mib, 4, nullptr, &len, nullptr, 0) == 0)
```

Comment:
this looks like it's the same as above? Why not just put the #if around the `mib` ?